### PR TITLE
Support latest signature of fuse_lowlevel_notify_expire_entry

### PR DESCRIPTION
**Pull request by HereThereBeDragons**

### CODE REVIEW

**Comment by HereThereBeDragons** [comment]
@cernvm-bot cloudtest

File: cvmfs/fuse_evict.h [hunk 1]
```diff
@@ -61,10 +62,18 @@ class FuseInvalidator : SingleCopy {
...
                   bool fuse_notify_invalidation);
+  
+  /**
+   * CONSTRUCTOR ONLY FOR UNITTESTS - mountpoint will illegally be null
+   * ( we do not want to construct a full mountpoint in the unittest )
+   */
+  FuseInvalidator(glue::InodeTracker *inode_tracker,
+                glue::DentryTracker *dentry_tracker,
+                void **fuse_channel_or_session,
+                bool fuse_notify_invalidation);
```
Comment (jblomer) [inline cvmfs/fuse_evict.h]: Since we already define the friend tests, can we make this constructor private?

File: cvmfs/fuse_evict.h [hunk 2]
```diff
@@ -11,6 +11,7 @@
 #include "bigvector.h"
 #include "duplex_fuse.h"
 #include "gtest/gtest_prod.h"
+#include "mountpoint.h"
```
Comment (jblomer) [inline cvmfs/fuse_evict.h]: Should be enough to include in the compilation unit. Here, we can forward declare the `MounPoint` class.

File: cvmfs/fuse_evict.cc [hunk 3]
```diff
@@ -65,12 +65,33 @@ bool FuseInvalidator::HasFuseNotifyInval() {
...
+  g_fuse_notify_invalidation_ = fuse_notify_invalidation;
+  MakePipe(pipe_ctrl_);
+  memset(&thread_invalidator_, 0, sizeof(thread_invalidator_));
+  atomic_init32(&terminated_);
+}
+
+/**
+ * CONSTRUCTOR ONLY FOR UNITTESTS - mountpoint will illegally be null
+ * ( we do not want to construct a full mountpoint in the unittest )
+*/
```
Comment (jblomer) [inline cvmfs/fuse_evict.cc]: Comment only once, either here or in the header.

File: cvmfs/fuse_evict.cc [hunk 4]
```diff
@@ -236,6 +256,26 @@ void *FuseInvalidator::MainInvalidator(void *data) {
     uint64_t entry_parent;
     NameString entry_name;
     i = 0;
+
+#if CVMFS_USE_LIBFUSE == 2
+    int (*notify_func)(struct fuse_chan*, fuse_ino_t, const char*, size_t);
+    notify_func = &fuse_lowlevel_notify_inval_entry;
+#else
+
```
Comment (jblomer) [inline cvmfs/fuse_evict.cc]:
```suggestion
```

File: cvmfs/fuse_evict.cc [hunk 5]
```diff
@@ -236,6 +256,26 @@ void *FuseInvalidator::MainInvalidator(void *data) {
...
     i = 0;
+
+#if CVMFS_USE_LIBFUSE == 2
+    int (*notify_func)(struct fuse_chan*, fuse_ino_t, const char*, size_t);
+    notify_func = &fuse_lowlevel_notify_inval_entry;
+#else
+    
+    int (*notify_func)(struct fuse_session*, fuse_ino_t, const char*, size_t);
+    notify_func = &fuse_lowlevel_notify_inval_entry;
+
```
Comment (jblomer) [inline cvmfs/fuse_evict.cc]:
```suggestion
```

File: test/src/702-symlink_caching/main [hunk 6]
```diff
@@ -119,14 +121,18 @@ check_libfuse2() {
...
+
 }
 
 # for fuse3 (prerequirement: fuse >= 3.10)
 # checks that
 # 1) symlinks are cached in kernel
 # 2) symlinks switching from location with a short path --> long path should 
 #    not trunacte
+#
+# TODO TODO
```
Comment (jblomer) [inline test/src/702-symlink_caching/main]:
```suggestion
# TODO(HereThereBeDragons)
```

**Review by jblomer — CHANGES_REQUESTED** [review]
Great! Some minor comments.

**Comment by HereThereBeDragons** [comment]
@cernvm-bot cloudtest

**Review by jblomer — APPROVED** [review]

**Comment by HereThereBeDragons** [comment]
Related issue #2949

**Comment by vvolkl** [comment]
There's a build failure on debian 12: https://lcgapp-services.cern.ch/cvmfs-jenkins/job/CvmfsFullBuildDocker/CVMFS_BUILD_ARCH=docker-x86_64,CVMFS_BUILD_PLATFORM=debian12/4049/console

**Comment by jblomer** [comment]
Is this one of the buggy fuse versions?

**Comment by HereThereBeDragons** [comment]
@cernvm-bot unittest

**Comment by HereThereBeDragons** [comment]
@cernvm-bot cloudtest

**Review by vvolkl — APPROVED** [review]

**Pull request merged**